packages/server/postgres/migrations/2025-12-12T23:00:00.000Z_addOAuthProvider.ts

```diff
@@ -0,0 +1,74 @@
+import {type Kysely, sql} from 'kysely'
+
+export async function up(db: Kysely<any>): Promise<void> {
+  await db.schema
+    .createType('OAuthScopeEnum')
+    .asEnum(['graphql:query', 'graphql:mutation'])
+    .execute()
+
+  await db.schema
+    .createTable('OAuthAPIProvider')
+    .addColumn('id', 'serial', (col) => col.primaryKey())
+    .addColumn('orgId', 'varchar(100)', (col) =>
+      col.notNull().references('Organization.id').onDelete('cascade')
+    )
+    .addColumn('name', 'varchar(255)', (col) => col.notNull())
+    .addColumn('clientId', 'varchar(255)', (col) => col.notNull().unique())
+    .addColumn('clientSecret', 'varchar(43)', (col) => col.notNull())
+    .addColumn('redirectUris', sql`varchar(2048)[]`, (col) => col.notNull())
+    .addColumn('scopes', sql`"OAuthScopeEnum"[]`, (col) => col.notNull())
+    .addColumn('createdAt', 'timestamptz', (col) => col.defaultTo(sql`now()`).notNull())
+    .addColumn('updatedAt', 'timestamptz', (col) => col.defaultTo(sql`now()`).notNull())
+    .execute()
+
+  await sql`
+    CREATE TRIGGER update_oauth_api_provider_updated_at
+    BEFORE UPDATE ON "OAuthAPIProvider"
+    FOR EACH ROW
+    EXECUTE FUNCTION "set_updatedAt"();
+  `.execute(db)
+
+  await db.schema
+    .createIndex('idx_oauth_provider_org_id')
+    .on('OAuthAPIProvider')
+    .column('orgId')
+    .execute()
+
+  await db.schema
+    .createTable('OAuthAPICode')
+    .addColumn('id', 'serial', (col) => col.primaryKey())
```

Comment:
-2 I think your change went too far. It makes sense for `OAuthAPIProvider` to have a serial integer for an ID because it doesn't need to be cryptographically secure. We can hide how many rows are in our DB by using a fesitel cipher on it & that works pretty well, but it is possible to brute-force because that cipher is not cryptographically secure.  OAuthAPICode's code needs 128 bits of entropy because if someone cracks a code, it's a lot more serious than just knowing the number of rows in our DB table.